python/setup.py

```diff
@@ -1,85 +1,39 @@
-            os.path.dirname(self.get_ext_fullpath(ext.name)))
-
-        abi_flag = os.environ.get('CMAKE_CXX11_ABI', 'ON')
-        build_type = 'Debug' if self.debug else 'Release'
-        cmake_args = ['cmake',
-                      ext.sourcedir,
-                      '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + output_dir,
-                      '-DCMAKE_BUILD_TYPE=' + build_type,
-                      '-DCMAKE_CXX11_ABI=' + abi_flag]
-        cmake_args.extend(
-            [x for x in os.environ.get('CMAKE_COMMON_VARIABLES', '').split(' ')
-             if x])
-
-        env = os.environ.copy()
-        if not os.path.exists(self.build_temp):
-            os.makedirs(self.build_temp)
-        subprocess.check_call(cmake_args,
-                              cwd=self.build_temp,
-                              env=env)
-        subprocess.check_call(['make', '-j', ext.name],
-                              cwd=self.build_temp,
-                              env=env)
-        print()
-
-
-shutil.rmtree('build', ignore_errors=True)
-
 install_requires = []
 
-with open('../LICENSE', encoding='UTF-8') as f:
-    license_text = f.read()
-
 cuda_version = ''.join(os.environ.get('CUDA_VERSION', 'unknown')
                          .split('.')[:2])
 name = 'nvstrings-cuda{}'.format(cuda_version)
 version = os.environ.get('GIT_DESCRIBE_TAG', '0.0.0.dev0').lstrip('v')
-setup(name=name,
-      description='CUDA strings Python bindings',
-      version=version,
-      py_modules=['nvstrings', 'nvcategory'],
-      url='https://github.com/rapidsai/custrings',
-      author='NVIDIA Corporation',
-      license=license_text,
-      install_requires=install_requires,
-      ext_modules=[CMakeExtension('NVStrings', '../cpp'),
-                   CMakeExtension('pyniNVStrings', '../cpp'),
-                   CMakeExtension('NVCategory', '../cpp'),
-                   CMakeExtension('pyniNVCategory', '../cpp'),
-                   CMakeExtension('NVText', '../cpp'),
-                   CMakeExtension('pyniNVText', '../cpp')],
-      cmdclass={'build_ext': CMakeBuildExt},
-      headers=['../cpp/include/NVStrings.h',
-               '../cpp/include/NVCategory.h',
-               '../cpp/include/NVText.h'],
-      zip_safe=False
-      )
+setup(
+    name=name,
+    description='CUDA strings Python bindings',
+    url='https://github.com/rapidsai/custrings',
+    version=version,
+    classifiers=[
+        "Intended Audience :: Developers",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7"
+    ],
+    py_modules=['nvstrings', 'nvcategory', 'nvtext'],
+    author='NVIDIA Corporation',
+    license="Apache",
+    install_requires=install_requires,
+    data_files=[('', ['pyniNVStrings.so', 'pyniNVCategory.so', 'pyniNVText.so'])],
+    packages=[''],
+    package_data={
+        '': [
+            'pyniNVStrings.so', 'pyniNVCategory.so', 'pyniNVText.so'
+        ],
```

Comment:
So I haven't dug into this deeply yet. Apologies if I'm asking a question with an obvious answer.

That said, normally when building Python extension modules, I use `ext_modules` to list some things I'd like to build and let `distutils` do the heavy lifting. Would be interested to know why we are using a different approach.

Should add I wouldn't be surprised by the need for a different approach. Just am looking for context.

Reply:
Yea, talking offline with @raydouglass and it seems like building these as part of the Python install process instead of the typical CMake process is probably the best bet.

Reply:
We might be able to get this to work by adding `package_dir={'': '.'}`.